### PR TITLE
feat: allow specifying version in config [APE-1473]

### PR DESCRIPTION
**Pull request by antazoey**
### What I did

same as https://github.com/ApeWorX/ape-solidity/pull/46

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

File: tests/projects/version_in_config/ape-config.yaml
Comment (sabotagebeats) [inline tests/projects/version_in_config/ape-config.yaml:1]:
```
$ ape compile -f
INFO: Compiling 'Token.vy'.
ERROR: (VyperCompileError) Token.vy
VersionException:line 1:0 Version specification "0.3.9" is not compatible with compiler version "0.3.7"
$ cat contracts/Token.vy | grep "@version"
# @version 0.3.9
$ cat ape-config.yaml
name: TokenProject
plugins:
  - name: vyper
vyper:
  version: 0.3.7
$ pico ape-config.yaml
$ cat ape-config.yaml
name: TokenProject
plugins:
  - name: vyper
vyper:
  version: 0.3.9
$ ape compile -f
INFO: Compiling 'Token.vy'.
```
Note: This seems to work for me, when the pragma matches the config it works and when it doesn't it fails with the error. Is this the desired effect?

**Review by sabotagebeats — APPROVED** [review]
This seems to work, can you please review comment to make sure that it is appropriately working for me the way that it is desired to work?

File: tests/projects/version_in_config/ape-config.yaml
Comment (Ninjagod1251) [reply tests/projects/version_in_config/ape-config.yaml:1]: I can reproduce the same result. I just want to make sure this is what we want @antazoey

Comment (sabotagebeats) [reply tests/projects/version_in_config/ape-config.yaml:1]: also: 
if i do the opposite, where they are not the same it also fails in the same manner: 
```
$ ape compile -f
INFO: Compiling 'Token.vy'.
ERROR: (VyperCompileError) Token.vy
VersionException:line 1:0 Version specification "0.3.7" is not compatible with compiler version "0.3.9"
```

File: tests/test_compiler.py [hunk 1]
```diff
@@ -445,3 +445,17 @@ def test_trace_source_default_method(geth_provider, account, project):
...
+    version_from_config = "0.3.7"
+    spy = mocker.patch("ape_vyper.compiler.vvm_compile_standard")
+    with config.using_project(path) as project:
+        contract = project.contracts_folder / "v_contract.vy"
+        settings = compiler.get_compiler_settings((contract,))
+        assert str(list(settings.keys())[0]) == version_from_config
+
+        # Show it uses this version in the compiler.
+        project.load_contracts(use_cache=False)
+        assert str(spy.call_args[1]["vyper_version"]) == version_from_config
```
Comment (Ninjagod1251) [inline tests/test_compiler.py:461]: I have the same results as @sabotagebeats i just want to make sure it works as in 10 digs

**Review by Ninjagod1251 — APPROVED** [review]

File: tests/projects/version_in_config/ape-config.yaml
Comment (sabotagebeats) [reply tests/projects/version_in_config/ape-config.yaml:1]: as requested I have tested it without the pragma now and it still compiles: 

```
$ cat ape-config.yaml
name: TokenProject
plugins:
  - name: vyper
vyper:
  version: 0.3.9
$ cat contracts/Token.vy | grep "@version"
$ ape compile -f
INFO: Compiling 'Token.vy'.
```

Comment (sabotagebeats) [reply tests/projects/version_in_config/ape-config.yaml:1]: it also works if there is no pragma and no version in the config: 
```
$ cat ape-config.yaml
name: TokenProject
plugins:
  - name: vyper
$ cat contracts/Token.vy | grep "@version"
$ ape compile -f
INFO: Compiling 'Token.vy'.
```

Comment (sabotagebeats) [reply tests/projects/version_in_config/ape-config.yaml:1]: it also works if there is only a pragma and there is not a version in the config: 

```
$ cat ape-config.yaml
name: TokenProject
plugins:
  - name: vyper
$ cat contracts/Token.vy | grep "@version"
# @version 0.3.7
$ ape compile -f
INFO: Compiling 'Token.vy'.
```

**Pull request merged**